core/consensus/slots/src/lib.rs

```diff
@@ -51,7 +51,8 @@ pub trait SlotWorker<B: Block> {
 	type OnSlot: IntoFuture<Item = (), Error = consensus_common::Error>;
 
 	/// Called when the proposer starts.
-	fn on_start(&self, slot_duration: u64) -> Result<(), consensus_common::Error>;
+	#[deprecated(note = "Not called. Please perform any initialization before calling start_slot_worker.")]
```

Comment:
I think we can just remove this.  `master` has no guarantee of backwards-compatibility.  That’s what I did with inherent digests.

Reply:
Since this PR breaks more stuff, I'll just remove it.

Reply:
Oh well, I didn't have time.